include/Gaffer/SplinePlug.h

```diff
@@ -72,10 +72,20 @@ class SplinePlug : public ValuePlug
 		IECORE_RUNTIMETYPED_DECLARETEMPLATE( SplinePlug<T>, ValuePlug );
 		IE_CORE_DECLARERUNTIMETYPEDDESCRIPTION( SplinePlug<T> );
 
+		// TODO - this is where John had suggested this go in his initial email, but does it
+		// really make for this enum to be part of each separate templated version of SplinePlug?
+		enum Interpolation
+		{
+			Linear,
+			CatmullRom,
+			BSpline,
+		};
+
 		SplinePlug(
 			const std::string &name = defaultName<SplinePlug>(),
 			Direction direction=In,
-			const T &defaultValue = T(),
+			const typename T::PointContainer &defaultPoints = T::PointContainer(),
```

Comment:
If we were to do this SplinePlug::ValueType struct, would the SplinePlug still be templated on an IECore::Spline?

It kind of seems like it could just be templated on the Color3f or float, and then spline() would return IECore::Spline<float, T>.  If we're overhauling everything, it seems a bit weird to keep IECore::Spline as the thing it's templated on, if it's just one member function that does anything with IECore::Spline.